lib/SILGen/SILGenPoly.cpp

```diff
@@ -1360,7 +1360,7 @@ namespace {
             output = SGF.emitManagedRetain(Loc, output.getValue());
           }
 
-          if (output.getOwnershipKind() == ValueOwnershipKind::Owned) {
+          if (output.getOwnershipKind() == ValueOwnershipKind::Guaranteed) {
```

Comment:
Hmmm... this switch case looks incorrect to me. What /should/ be happening here is that if we are trying to pass a value to in_guaranteed but have something that is owned, we perform a borrow.  If we are trying to perform an in, but we have a guaranteed value, we perform a copy.